src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsWithPatternAction.java

```diff
@@ -0,0 +1,175 @@
+package pl.damianszczepanik.jenkins.buildhistorymanager.model.actions;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import org.jenkinsci.remoting.RoleChecker;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import hudson.FilePath;
+import hudson.FilePath.FileCallable;
+import hudson.Util;
+import hudson.model.Run;
+import hudson.remoting.VirtualChannel;
+import jenkins.util.VirtualFile;
+
+/**
+ * Deletes the artifacts with patterns.
+ */
+public class DeleteArtifactsWithPatternAction extends Action {
+    private static final Logger LOGGER = Logger.getLogger(DeleteArtifactsWithPatternAction.class.getName());
+
+    private String include;
+    private String exclude;
+
+    public String getInclude() {
+        return include;
+    }
+
+    @DataBoundSetter
+    public void setInclude(String include) {
+        this.include = include;
+    }
+
+    public String getExclude() {
+        return exclude;
+    }
+
+    @DataBoundSetter
+    public void setExclude(String exclude) {
+        this.exclude = exclude;
+    }
+
+    @DataBoundConstructor
+    public DeleteArtifactsWithPatternAction() {
+        // Jenkins stapler requires to have public constructor with @DataBoundConstructor
+    }
+
+    public static String removeLastSlash(String archiveRootPath) {
+        if(archiveRootPath.endsWith("/")) {
+            return archiveRootPath.substring(0, archiveRootPath.lastIndexOf("/"));
+        } else {
+            return archiveRootPath;
+        }
+    }
+
+    public static boolean isDirEmpty(Path path) throws IOException {
+        if (Files.isDirectory(path)) {
+            try (Stream<Path> entries = Files.list(path)) {
+                return !entries.findFirst().isPresent();
+            }
+        }        
+        return false;
+    }
+
+    // if 'file' is on a different node, this FileCallable will be transferred to that node and executed there.
+    public static final class Delete implements FileCallable<Void> {
+        private static final long serialVersionUID = 1;
```

Comment:
needed ? 1 does not look like valid serial version id

Reply:
Yes. 'FileCallable' must be serializable. 
https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html

Reply:
Then make it properly https://www.google.com/search?q=generate+serialVersionUID

Reply:
I added 1L .